### PR TITLE
Switch k8s metrics (container, pod, node, namespace) to use pdata.

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Switch k8s metrics (container, pod, node, namespace) to use pdata.

### CODE REVIEW

File: receiver/k8sclusterreceiver/internal/container/containers.go [hunk 1]
```diff
@@ -31,127 +30,40 @@ const (
...
-					Timeseries: []*metricspb.TimeSeries{
-						val,
-					},
-				},
-			)
+func GetSpecMetrics(set receiver.CreateSettings, c corev1.Container, pod *corev1.Pod) pmetric.Metrics {
+	mb := imetadata.NewMetricsBuilder(imetadata.DefaultMetricsBuilderConfig(), set)
+	ts := pcommon.NewTimestampFromTime(time.Now())
+	mb.RecordK8sContainerCPURequestDataPoint(ts, float64(c.Resources.Requests.Cpu().MilliValue())/1000.0)
+	mb.RecordK8sContainerCPULimitDataPoint(ts, float64(c.Resources.Limits.Cpu().MilliValue())/1000.0)
```
Comment (dmitryax) [inline receiver/k8sclusterreceiver/internal/container/containers.go:39]: We miss `k8s.container.memory_request` and `k8s.container.memory_limit` now

File: receiver/k8sclusterreceiver/internal/node/nodes.go [hunk 2]
```diff
@@ -57,79 +48,54 @@ func Transform(node *corev1.Node) *corev1.Node {
...
+		switch v1NodeAllocatableTypeValue {
+		case corev1.ResourceCPU:
 			// cpu metrics must be of the double type to adhere to opentelemetry system.cpu metric specifications
-			val = utils.GetDoubleTimeSeries(float64(quantity.MilliValue()) / 1000.0)
-			valType = metricspb.MetricDescriptor_GAUGE_DOUBLE
+			mb.RecordK8sNodeAllocatableCPUDataPoint(ts, float64(quantity.MilliValue())/1000.0)
+		case corev1.ResourceMemory:
+			mb.RecordK8sNodeAllocatableMemoryDataPoint(ts, quantity.Value())
+		case corev1.ResourceEphemeralStorage:
+			mb.RecordK8sNodeAllocatableEphemeralStorageDataPoint(ts, quantity.Value())
```
Comment (dmitryax) [inline receiver/k8sclusterreceiver/internal/node/nodes.go:94]: `storage` type is missed here

**Comment by dmitryax** [comment]
Maybe better split them as before?

**Comment by atoulme** [comment]
Splitting in #23438 and #23437 for node and namespace respectively. I cannot split container and pod as easily.

**Pull request closed without merging**